### ISSUE TITLE
Change how templates with subdirectories work in Phoenix

### ISSUE THREAD

**Issue by josevalim** [phoenixframework/phoenix#416]
Today Phoenix includes all templates in a directory and subdirectories in a view. For example, if I have:

```
web/templates/user/index.html.eex
web/templates/user/comment/index.html.eex
```

The MyApp.UserView will include both "index.html.eex" and "comment/index.html.eex". I would like to change the current behaviour to include only the entries in the current directory.

In case we want to support subdirectories, I can think of two options:
1. Make each subdirectory opt-in:
   
   ```
   defmodule MyApp.UserView do
     use MyApp.View
     include_dir "comments"
   end
   ```
2. Make a distinction between upper case and lower case directories (we have discussed this in the past). For example, `MyApp.UserView` would include "web/templates/User". `MyApp.User.CommentView` would include "web/templates/User/Comment". Therefore, if we have "web/templates/User/comment/index.html.eex", it will be automatically included in the UserView because the directory starts in lowercase

Given 1. is considerably shorter to explain than 2., my vote is for 1. Other suggestions? Any preference?

**Comment by aforward** [comment]
For those landing here from a search, I believe this is implemented in #416, not #414

**Comment by aforward** [comment]
I enounctered this through search, and I believe `include_dir` was not implemented and the correct approach is to 

# A) Do not use nested, and just create a new view

It's only a few files of code, and you can render it using something like

```
<%= render(HelloWeb.MyNestedDirView, "test.html", message: "Assigns has an @.") %>
```

# B) Or, you can expand the search for nested directories

The new `pattern: "**/*",` line below helps you search subdirectories.

```
  def view do
    quote do
      use Phoenix.View,
        root: "lib/techblog_web/templates",
        pattern: "**/*",  # <---- add this to compile all
        namespace: TechblogWeb

      # Import convenience functions from controllers
      import Phoenix.Controller, only: [get_flash: 1, get_flash: 2, view_module: 1]

      # Include shared imports and aliases for views
      unquote(view_helpers())
    end
  end
```
